src/coreclr/System.Private.CoreLib/src/System/Runtime/InteropServices/CustomMarshalers/EnumeratorToEnumVariantMarshaler.cs

```diff
@@ -52,10 +47,7 @@ public IntPtr MarshalManagedToNative(object ManagedObj)
 
         public object MarshalNativeToManaged(IntPtr pNativeData)
         {
-            if (pNativeData == IntPtr.Zero)
-            {
-                throw new ArgumentNullException(nameof(pNativeData));
-            }
+            ArgumentNullException.ThrowIfNull(pNativeData);
```

Comment:
Why did you not used `IntPtr pNativeData!!` syntax?

Reply:
> Why did you not used IntPtr pNativeData!! syntax?

It doesn't work for IntPtr.
`error CS8992: Parameter 'IntPtr' is a non-nullable value type and cannot be null-checked.`
Something the language could consider.
cc: @RikkiGibson

Reply:
I wasn't sure what the behavior would be here. I think `intPtr == null` is always false, but I might be misunderstanding something. [SharpLab](https://sharplab.io/#v2:EYLgtghglgdgPgAQEwEYCwAoBBmABM3AYVwG9NcL88EAWXAWQAoBJGAFwAU2AnXADwCUpcpVGEA9jADO4gDYBTAHQB1blDbzGfXAF4duGAFdZs3AH5cKXCFxIBAbhEUAvpmdA===)

Reply:
> I wasn't sure what the behavior would be here. I think intPtr == null is always false, but I might be misunderstanding something.

Some APIs accept `IntPtr` instead of `void*`, so the check would be `ptr == IntPtr.Zero`.  I don't know whether it's common enough to really warrant language attention, though.  Corelib has a few APIs that do it, and so I added an internal ThrowIfNull overload to be able to use that in such cases, but I don't expect we'll be encouraging adding many new such APIs, instead preferring to just use pointers, which `!!` already supports.